packages/router-plugin/src/core/code-splitter/compilers.ts

```diff
@@ -151,14 +152,30 @@ export function compileCodeSplitReferenceRoute(
               }
 
               if (t.isObjectExpression(routeOptions)) {
+                if (opts.deleteNodes && opts.deleteNodes.length > 0) {
+                  routeOptions.properties = routeOptions.properties.filter(
+                    (prop) => {
+                      if (t.isObjectProperty(prop)) {
+                        if (t.isIdentifier(prop.key)) {
+                          if (
+                            opts.deleteNodes?.includes(prop.key.name as any)
```

Comment:
Might we worth checking this out against a large amount of routes to see if there's much of perf impact of using an Array vs a Set.

Nothing blocking though since its currently just expected to `Array<'ssr'> | undefined`.